### ISSUE TITLE
Deriving Display on an enum triggers clippy::pattern_type_mismatch

### ISSUE THREAD

**Issue by jsim2010** [frozenlib/parse-display#8]
Running clippy on the following code:

```
#![deny(clippy::pattern_type_mismatch)]

#[derive(parse_display::Display)]
#[display(style = "CamelCase")]
enum Test {
    First,
    Second,
}
```

results in the following error

```
error: type of pattern does not match the expression type
   |
   |   enum Test {
   |  ______^
   | |     First,
   | |_________^
   |
   = help: use `*` to dereference the match expression or explicitly match against a `&_` pattern and adjust the enclosed variable bindings
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#pattern_type_mismatch
```

**Comment by frozenlib** [comment]
Thanks!

This issue was fixed in ef1f463398302731c905c91201f003fb0fdf503c.

**Comment by jsim2010** [comment]
Awesome! Thank you for the quick response.
